package idgs.admin.pb;

option java_outer_classname = "PbAdminService";

message AttributeRequest {
  required string attribute = 1; // attribute format should be xxx.xxx.xxx
}

enum ResponseStatus{
  Success = 1;
  Error = 2;
}

message AttributeResponse {
  required ResponseStatus status = 1;
  required string module_name = 2;
  required string attribute = 3; // attribute format should be xxx.xxx.xxx
  optional string value = 4; // if status is error, the value may be is error response
}

message ModuleOpRequest {
  required string module_name = 1;
  repeated AttributeRequest attributes = 2;
}

message ModuleOpResponse {
  required string module_name = 1;
  repeated AttributeResponse attributes = 2;
}

message AdminRequest {
  repeated ModuleOpRequest module_op_request = 1;
}

message AdminResponse {
  repeated ModuleOpResponse module_op_response = 1;
}